itou/job_applications/admin.py

```diff
@@ -162,7 +166,42 @@ class JobApplicationAdmin(admin.ModelAdmin):
         ),
     ]
 
-    search_fields = ("pk", "to_siae__siret", "job_seeker__email", "sender__email")
+    def get_search_fields(self, request):
+        # Without search_fields, the search bar is hidden.
+        # Provide a dummy value that’s quick to search, in order not to slow
+        # down relevant expensive searches that are added dynamically.
+        search_fields = ["state__startswith"]
+
+        search_term = request.GET.get("q", "").strip()
+        try:
+            uuid.UUID(search_term)
+        except (TypeError, ValueError):
+            pass
+        else:
+            search_fields.append("pk__exact")
+        siren_length = 9
+        siret_length = 14
+        if search_term.isdecimal() and len(search_term) in [siren_length, siret_length]:
+            search_fields.append("to_siae__siret__exact")
+        return search_fields
```

Comment:
Vu que je doute qu'on veuille chercher par état je pense qu'on peux virer ce champ si on tombe dans un des autres cas, surtout que si je comprend bien comment ça fonctionne on se retrouve à finalement chercher quelque chose qui ne nous sert pas pour juste afficher la barre de recherche.

Et pour que le SIREN match faudrait pas remplacer le `siret__exact` par `siret__startswith` ?

```suggestion
        search_fields = []

        search_term = request.GET.get("q", "").strip()
        try:
            uuid.UUID(search_term)
        except (TypeError, ValueError):
            pass
        else:
            search_fields.append("pk__exact")
        siren_length = 9
        siret_length = 14
        if search_term.isdecimal() and len(search_term) in [siren_length, siret_length]:
            search_fields.append("to_siae__siret__startswith")
        return search_fields or ["state__startswith"]
```

Reply:
Très exact, merci !